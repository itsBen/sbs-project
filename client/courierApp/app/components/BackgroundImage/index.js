import React from 'react';
import { ImageBackground } from 'react-native';

export default ({ children, imageSource }) => {
  return (
    <ImageBackground
      source={
        imageSource ? require(imageSource) : require('../../assets/Piglet.jpg')
      }
      style={backgroundImageStyle}
    >
      {children}
    </ImageBackground>
  );
};

const backgroundImageStyle = {
  flex: 1,
  width: null,
  height: null,
};
