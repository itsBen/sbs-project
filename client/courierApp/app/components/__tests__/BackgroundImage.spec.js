/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BackgroundImage from '../BackgroundImage';

it('renders correctly', () => {
  const tree = renderer.create(<BackgroundImage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with different image', () => {
  const tree = renderer
    .create(
      <BackgroundImage imageSource="../../assets/forever-lost-test.jpg" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
