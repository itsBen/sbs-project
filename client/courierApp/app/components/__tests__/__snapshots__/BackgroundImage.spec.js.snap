// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`renders correctly with different image 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
</View>
`;
